client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { pick } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_UPDATE_POSTER } from 'state/action-types';
+import {
+	setPosterUrl,
+	showError,
+	showUploadProgress,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updatePoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params || 'atTime' in action.params ) ) {
+		return next( action );
+	}
+
+	const { atTime, file } = pick( action.params, [ 'atTime', 'file' ] );
+	const params = Object.assign(
+		{
+			apiVersion: '1.1',
+			method: 'POST',
+			path: `/videos/${ action.videoId }/poster`,
+		},
+		file && { formData: [ [ 'poster', file ] ] },
+		atTime !== undefined && { body: { at_time: atTime } },
+	);
+
+	dispatch( http( params, action ) );
+
+	return next( action );
+};
+
+export const updatePosterUrl = ( { dispatch }, action, next, { poster } ) => {
+	dispatch( setPosterUrl( poster ) );
+};
+
+export const updatePosterError = ( { dispatch } ) => {
+	dispatch( showError() );
```

Comment:
We do actually surface an error message to the user in the case where the poster could not be updated. To be clear, the new poster URL gets generated server-side and is sent back to the client on a successful update. If the poster updates without issue, the video editor modal closes and goes back to the Media Library modal, which will load the VideoPress player with the new poster showing.

In the event of an error occurring when updating a poster, we're not in a situation where the poster URL needs to be reverted because it was never changed in the first place.

As for the RESET_STATE action, that's there to clear out any existing state when opening the video editor modal again. Otherwise you might open it and see an error message if your first attempt at updating the poster wasn't successful.